pkg/local-disk-manager/handler/localdiskvolume/localdiskvolume.go

```diff
@@ -459,3 +469,56 @@ func (v *DiskVolumeHandler) GetBoundDisk() string {
 func (v *DiskVolumeHandler) For(volume *v1alpha1.LocalDiskVolume) {
 	v.Ldv = volume
 }
+
+// CanSafelyMount returns true if the volume is safe to be mounted, false otherwise.
+// case 1: if there are multiple disks having the same identifier, return false to prevent mount the wrong disk
+func (v *DiskVolumeHandler) CanSafelyMount() (bool, error) {
+	// this can not be happened, stop the mount process
+	if len(v.Ldv.Status.DevLinks) == 0 {
+		return false, fmt.Errorf("no devlinks found for volume %s", v.Ldv.Name)
+	}
+
+	// skip if no id_path found
+	if len(v.Ldv.Status.DevLinks[v1alpha1.LinkByID]) == 0 {
+		return true, nil
+	}
+
+	// NOTE: don't use id_path here because it might be different from the dev-link used by the volume
+	// 1. get the symbol link that actually used by the volume
+	deviceLink, err := getDeviceLinkByVolume(v.Ldv.Status.VolumePath)
+	if err != nil {
+		return false, err
+	}
+
+	// 2. list all devices and check whether this link exists at 2 or more different devices
+	allDevices, err := udev.ListAllBlockDevices()
+	if err != nil {
+		return false, err
+	}
+
+	var matchedDevices []manager.Attribute
+	for i, device := range allDevices {
```

Comment:
Maybe we can add serial number to ldv status, and you can match serial number here. It's right if there is only one device whose serial number match ldv's serial number.